src/cli.rs

```diff
@@ -1,14 +1,195 @@
 };
 
+use inflector::Inflector;
+use quote::{format_ident, quote};
+
+use crate::config::{find_config_file, load_config_file, ParsedConfigFile};
+
+const PER_PROJECT_DIRECTORY_NAME: &str = ".tree-sitter-lint";
+
+const LOCAL_BINARY_PROJECT_NAME: &str = "tree-sitter-lint-local";
+
 pub fn bootstrap_cli() {
-    let path_to_local_binary =
-        "/Users/jrosse/prj/tree-sitter-lint/.tree-sitter-lint/tree-sitter-lint-local/target/release/tree-sitter-lint-local";
-    let mut handle = Command::new(path_to_local_binary)
+    let config_file_path = find_config_file();
+    let project_directory = config_file_path.parent().unwrap();
+    let per_project_directory = project_directory.join(PER_PROJECT_DIRECTORY_NAME);
+    let local_binary_project_directory = per_project_directory.join(LOCAL_BINARY_PROJECT_NAME);
+    let path_to_local_release_binary =
+        local_binary_project_directory.join(format!("target/release/{LOCAL_BINARY_PROJECT_NAME}"));
+    if should_regenerate_local_binary(&config_file_path, &path_to_local_release_binary) {
+        regenerate_local_binary(&local_binary_project_directory, &Path::new("..").join(".."));
+    }
+    let mut handle = Command::new(path_to_local_release_binary)
         .args(env::args_os().skip(1))
         .spawn()
         .unwrap();
     process::exit(handle.wait().unwrap().code().unwrap_or(1));
 }
+
+fn should_regenerate_local_binary(
+    config_file_path: &Path,
+    path_to_local_release_binary: &Path,
+) -> bool {
+    let local_release_binary_modified_timestamp = match path_to_local_release_binary
+        .metadata()
+        .ok()
+        .and_then(|metadata| metadata.modified().ok())
+    {
+        None => return true,
+        Some(timestamp) => timestamp,
+    };
+    let config_file_modified_timestamp = config_file_path
+        .metadata()
+        .expect("Couldn't read config file metadata")
+        .modified()
+        .expect("Couldn't get config file modified timestamp");
+    config_file_modified_timestamp > local_release_binary_modified_timestamp
+}
+
+const LOCAL_RULES_DIR_NAME: &str = "local_rules";
+
+fn regenerate_local_binary(
+    local_binary_project_directory: &Path,
+    relative_path_from_local_binary_project_directory_to_project_directory: &Path,
+) {
+    eprintln!("Config changed, regenerating local binary");
+    let parsed_config_file = load_config_file();
+    let local_binary_project_src_directory = local_binary_project_directory.join("src");
+    let local_binary_project_cargo_toml_path = local_binary_project_directory.join("Cargo.toml");
+    if local_binary_project_directory.is_dir() {
+        let _ = fs::remove_dir_all(&local_binary_project_src_directory);
+        let _ = fs::remove_file(&local_binary_project_cargo_toml_path);
+        let _ = fs::remove_file(local_binary_project_directory.join("Cargo.lock"));
+    }
+    fs::create_dir_all(local_binary_project_directory)
+        .expect("Couldn't create local binary project directory");
```

Comment:
Oh I think I had this `.expect()` when I was fully wiping out the `tree-sitter-lint-local/` project directory before (vs now preserving it to be able to preserve `target/`)

So I'm actually surprised this isn't failing (I thought I read that `fs::create_dir_all()` would fail if the directory already existed)